prevent libgccjit.so download on unsupported os/arch

Skgland opened the pull request:
Alternative to #528, I find it easier to show what I meant rather than tell.

Fixing up the download URL while at it. 

fixes #527

GuillaumeGomez commented on build_system/src/config.rs:
```diff
@@ -469,11 +463,28 @@ impl ConfigInfo {
+    } else {
+        eprintln!(
+            r"\
+        pre-compiled libgccjit.so not available for this os or architecture\n\
+        please compile it yourself and in the config.toml set\n\
+        `download-gccjit = false` and `gcc-path` to the appropriate directory\
```
```suggestion
        Pre-compiled libgccjit.so not available for this os or architecture.\n\
        Please compile it yourself and update the `config.toml` file\n\
        to `download-gccjit = false` and set `gcc-path` to the appropriate directory.\
```

GuillaumeGomez commented on build_system/src/config.rs:
```diff
@@ -469,11 +463,28 @@ impl ConfigInfo {
+        format!("https://github.com/rust-lang/gcc/releases/download/master-{}/libgccjit.so", commit)
+    } else {
+        eprintln!(
+            r"\
```
Is the `r` useful here?

Skgland replied:
Quite the opposite of useful actually as a raw string is not what I want here.🤦‍♂️

GuillaumeGomez commented on build_system/src/config.rs:
```diff
@@ -469,11 +463,28 @@ impl ConfigInfo {
     with_progress_bar: bool,
 ) -> Result<(), String> {
+    let url = if std::env::consts::OS == "linux" || std::env::consts::ARCH == "x86_64" {
+        format!("https://github.com/rust-lang/gcc/releases/download/master-{}/libgccjit.so", commit)
```
Thanks! :)

Skgland replied:
Oh, looking at this I just noticed that I messed up the condition it should be and not or.

GuillaumeGomez replied:
Oh indeed! Completely missed it as well. ^^'

Skgland commented:
Ok, applied your suggestion regarding the message text change, fixed the conditional to be `&&` instead of `||` and changed the raw-string that shouldn't be raw to a normal string.
Assuming CI agrees, this is good to go from my point.

GuillaumeGomez commented on build_system/src/config.rs:
```diff
@@ -469,11 +463,28 @@ impl ConfigInfo {
+    } else {
+        eprintln!(
+            "\
+        Pre-compiled libgccjit.so not available for this os or architecture.\n\
+        Please compile it yourself and update the `config.toml` file\n\
+        to `download-gccjit = false` and set `gcc-path` to the appropriate directory.\
+        "
```
Just thinking but at this point we could simply do:

```suggestion
Pre-compiled libgccjit.so not available for this os or architecture.
Please compile it yourself and update the `config.toml` file
to `download-gccjit = false` and set `gcc-path` to the appropriate directory."
```

Skgland replied:
Sure, if that is preferred, done
